release/models/hashing/openconfig-hashing.yang

```diff
@@ -0,0 +1,238 @@
+module openconfig-hashing {
+    yang-version "1";
+
+    // namespace
+    namespace "http://openconfig.net/yang/hashing";
+
+    prefix "oc-hashing";
+
+    // import some basic types
+    import openconfig-extensions { prefix oc-ext; }
+
+    // meta
+    organization "OpenConfig working group";
+
+    contact
+      "OpenConfig working group
+      netopenconfig@googlegroups.com";
+
+    description
+      "Model for managing hashing policies that would be referenced by the
+      interfaces model.";
+
+
+    oc-ext:openconfig-version "0.1.0";
+
+    revision "2023-08-08" {
+        description "Initial hashing model.";
+        reference "0.1.0";
+    }
+
+    // OpenConfig specific extensions for module metadata.
+    oc-ext:regexp-posix;
+    oc-ext:catalog-organization "openconfig";
+    oc-ext:origin "openconfig";
+
+    typedef hash-field-type-ipv4 {
+        type enumeration {
+            enum NONE {
+                description "Do not use any IPv4 fields in the
+                    calculation of the hash.";
+            }
+            enum SRC_ADDRESS {
+                description "Use only the IPv4 source address in the
+                    calculation of the hash.";
+            }
+            enum DST_ADDRESS {
+                description "Use only the IPv4 destination address in the
+                    calculation of the hash.";
+            }
+            enum BOTH {
+                description "Use both the IPv4 source and
+                    destination addresses in the calculation of the
+                    hash.";
+            }
+            enum TARGET_DEFINED {
+                description "Use the default fields that are used by
+                    the device for calculating the hash.";
+            }
+        }
+        description "The IPv4 fields that should be used to
+            compute the hash.";
+    }
+
+    typedef hash-field-type-ipv6 {
```

Comment:
Ditto the considerations for the IPv4 typedef.

Reply:
See proposal in #959